jupyter_events/schema.py

```diff
@@ -51,30 +59,65 @@ def __init__(
     def __repr__(self):
         return json.dumps(self._schema, indent=2)
 
+    @staticmethod
+    def _ensure_yaml_loaded(schema, was_str=False) -> None:
+        """Ensures schema was correctly loaded into a dictionary. Raises
+        EventSchemaLoadingError otherwise."""
+        if isinstance(schema, dict):
+            return
+
+        error_msg = "Could not deserialize schema into a dictionary."
+
+        def intended_as_path(schema):
+            path = Path(schema)
+            return path.match("*.yml") or path.match("*.yaml") or path.match("*.json")
```

Comment:
I think this is the best way of doing this since Pathlib doesn't provide a magical "is path valid" method. But I'm open to suggestions on this one.

Reply:
I think this is good enough. Thanks, @dlqqq :+1: